src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -751,6 +739,11 @@ private bool EqualsComplicatedCases(NamedTypeSymbol other, TypeCompareKind compa
                 }
             }
 
+            if (this.IsTupleType && !this.TupleNamesEquals(other, comparison))
```

Comment:
>this.IsTupleType &&  [](start = 16, length = 20)

Is this check necessary?

Reply:
Yes, we can compare types considering or ignoring tuple names. That's where that logic is implemented.

---
In reply to: [355213367](https://github.com/dotnet/roslyn/pull/39370#discussion_r355213367) [](ancestors = 355213367)